addons/website_sale_loyalty/models/sale_order.py

```diff
@@ -175,9 +175,9 @@ def _get_free_shipping_lines(self):
         return self.order_line.filtered(lambda l: l.reward_id.reward_type == 'shipping')
 
     def _allow_nominative_programs(self):
-        if not request or not hasattr(request, 'website'):
-            return super()._allow_nominative_programs()
```

Comment:
For clarity puprose, could you let this code in that form?
1: case where this custom code is not needed, like when not from the website ==> just return super
2: specific case that requires custom codes, and not the generic one from the super or adding something

Reply:
Do you mean to write the code similar to this?

```
if not request or not hasattr(request, 'website'):
            return super()._allow_nominative_programs()
elif request and hasattr(request, 'website') and request.website.is_public_user():
            return self.state == 'sent' and super()._allow_nominative_programs()
```

I'm not sure how to interpret #2. Do you mean that in my custom case there is no need to return super?